libs/application-generic/src/usecases/create-workflow/create-workflow.usecase.ts

```diff
@@ -338,7 +380,14 @@ export class CreateWorkflow {
 
     this.sendTemplateCreationEvent(command, triggerIdentifier);
 
-    return item;
+    return this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        userId: command.userId,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        identifierOrInternalId: savedWorkflow._id,
+      }),
+    );
```

Comment:
Just fetch again using the `GetWorkflowByIds` use-case to let that use-case handle data collection merging in a single place. Not the most performant, but this is an infrequently called use-case so it's acceptable.